dotnet/src/SemanticKernel/Configuration/IBackendConfig.cs

```diff
@@ -10,4 +12,11 @@ public interface IBackendConfig
     /// An identifier used to map semantic functions to backend,
     /// decoupling prompts configurations from the actual model used.
     public string Label { get; }
+
+    /// <summary>
+    /// Creates a new instance of AI backend.
+    /// </summary>
+    /// <param name="kernel"></param>
+    /// <returns></returns>
+    public ITextCompletionClient CreateCompletitionAiBackEnd(IKernel kernel);
```

Comment:
this doesn't seem the right interface to add a builder. For instance we have 2 AI service types (completion and embedding generation) so if we add this, we should also add one method for the embeddings. But we're going to have many more (10-15 more) and we can be adding all of them here, forcing all the config classes to become also builder classes. In other words, the config classes are value objects, and their only responsibility is carrying configuration values, for the logic that instantiate each AI backend/client